### DISCUSSION TITLE
XIAOGYANやってみたよ！！ユーザー投稿

### DISCUSSION THREAD

**Discussion by matsujirushi** [algyan/XIAOGYAN#50] [General]
* XIAOGYAN、ゲットしたぜ！
* はんだ付け難しいなぁ
* うまく動いたぞ！

など、**みなさんの一喜一憂**を投稿くださいませ。

**Comment by pojiro** [comment]
MFK でもらった [#XIAOGYAN](https://twitter.com/hashtag/XIAOGYAN?src=hashtag_click) 基板を5/14（日）に実装しました！
HARDWARETEST プロジェクト（以下リンク）が用意されているので動作確認までスムーズにできました🕺
https://github.com/algyan/xiaogyan_examples/tree/main/platformio/HardwareTest

https://github.com/algyan/XIAOGYAN/assets/4096956/72fc56af-3f26-47a4-9992-2111e6e4354b

**Reply by matsujirushi** [reply]
すばらしい！
XIAO ESP32C3を直付け、、、思い切りがありますねｗ

ぜひ、AtomVMを動かしてみてください！！（わたし自身、どんな感じで動くのか気になる...）

**Reply by matsujirushi** [reply]
AtomVM、ESP32-C3対応しているようだ。
https://github.com/atomvm/AtomVM/blob/master/README.ESP32.Md#building-atomvm-for-esp32
> idf.py set-target esp32 # or esp32c3, esp32s2, esp32s3

**Reply by pojiro** [reply]
マニュアルに従っただけですが、AtomVM動きました！

![image](https://github.com/algyan/XIAOGYAN/assets/4096956/6676f185-78ed-4588-9a86-32352d10fd9d)

動かしたときのメモは[こちら](https://gist.github.com/pojiro/ccb9d88dedd0590a8ac6930b37633ee0)

**Reply by matsujirushi** [reply]
おぉ！
XIAOGYANドキュメントにリンク追加しますね。

**Comment by 610t** [comment]
むとう @610t と申します。

今、[XIAOGYANbitLess](https://github.com/610t/XIAOGYANbitLess) というものを作っています。
これは、XIAOGYANボードからScratch([Strech3](https://stretch3.github.io/))の[Micorbit More](https://microbit-more.github.io/index-ja.html)という拡張機能を使えるようにしたものです。
XIAOGYANボードがmicro:bitのように振るまいます。

現在、以下のような機能を実装しています。
- 5x5 LEDとしてのパターン表示
- 8x8 LEDへの文字列表示
- I2C加速度計 (ADXL345 or MPU6886)
- ボタンスイッチ
- スピーカーからのトーン音出力
- P0としてのアナログ入力(XIAOGYANのA3ポート)

まだ、ドキュメントなど足りないかと思いますが、興味があれば、使ってみてご意見いただけるとうれしいです。

**Reply by matsujirushi** [reply]
Scratchとの連携、すごいですね。

XIAOGYANドキュメントにリンク追加しますね。

**Reply by matsujirushi** [reply]
https://github.com/algyan/XIAOGYAN/commit/026e2c2f047c7cfd51a85930c73e55b3fa8fb658

**Comment by pscmps** [comment]
パスコンパスといいます。
私もMFKyotoでいただいたXIAOGYAN基板を先日作成しました！
表面実装の経験がほぼなくて不安だったのですが、チップ抵抗やコンデンサの位置が離れていたので隣にブリッジしたりせず、なんと一発で動作OKでした！
これなら今後自分でプリント基板を発注する時もDIPでない部品もいけるのかもと少し自信が付いたのでとても良い経験になりました。
また、自分で実装すると愛着も湧きますね。
何かコントローラとして使ってみたいなぁと思います。

**Reply by matsujirushi** [reply]
一発動作、すばらしい！

**Comment by lipoyang** [comment]
Speaker2クラスをnRF52840に移植してみました。(ソースはまだ粗削りです)
サンプルコード(PlayWave)の録音データが謎すぎて正しく動作してるのかよく分かりません(^^;)

- PWM出力は HwPWM0 を使用
- タイマ割り込みは NRF_TIMER2  を直接操作

https://github.com/algyan/XIAOGYAN/assets/2517317/d265b9d6-d58b-495b-ae53-15bae12648f2

**Reply by matsujirushi** [reply]
ちゃんと動作していますー👍
（音源は、、、うちの娘が滑り台を降りるときの奇声ですｗ）

そろそろ、ライブラリをXIAO切替ができるように見直さないといけないですね...
気にはしていますが、手がぜんぜん追い付いていない。

**Reply by lipoyang** [reply]
Speaker2.cpp の nRF52840対応のソース(案)です。
algyan/XIAOGYANのDiscussionsのほうにも書いておきます。

https://gist.github.com/lipoyang/4a923c1e3795488505eeb8658f3cf4b9

**Comment by lipoyang** [comment]
XIAOの裏面のバッテリー端子にLiPoバッテリーを接続してみました。
![1B](https://github.com/algyan/XIAOGYAN/assets/2517317/a02220d1-67ae-4e2c-bfe6-bfbd21f3ec4a)

あまり美しくありませんが、パッドに力がかかると剥離しやすいのでホットボンドで固めてます。
![2B](https://github.com/algyan/XIAOGYAN/assets/2517317/cbd4922f-87c3-4b17-92f4-b900b6562226)

**Comment by lipoyang** [comment]
単三形ニッケル水素電池を3.3Vに昇圧するモジュール(ENEBATT)を利用してGroveコネクタから給電してみました。
(ただしXIAOの3.3Vピン→5Vピンの向きにダイオードを入れるべきなのと、USB給電との衝突は避けるべきですが。)

![ENEBATT](https://github.com/algyan/XIAOGYAN/assets/2517317/076dddd3-e0f2-4a0e-bd32-326eeeeefcb2)

**Reply by matsujirushi** [reply]
メモ。

* ['ENEBATT' - 1Cell→3.3V昇圧モジュール](https://osafune.github.io/enebatt_jp.html)

**Comment by mnltake** [comment]
https://github.com/maxint-rd/TM16xx 
のライブラリを参考に文字を流してみました

https://gist.github.com/mnltake/e177ab18d4f5702b5dfce2d67a22d054


https://github.com/algyan/XIAOGYAN/assets/45388626/2dc38770-45b3-45f6-94c6-ae890b1b858a

**Comment by siroitori0413** [comment]
とり子といいます。
部品実装を終え無事に完成しました！
HardwareTestがすべて動作できるようになったので感無量です。

はんだづけの手順が綺麗にまとめてあったのでとても助かりました。

動作確認時に音が鳴らなくて愕然としたのですが、公開されているHardwareTestのPlatformIOのプログラムに少し手を入れてSerial.printさせてボタンの押下がとれているところまでが確認できたのでスピーカー周りを追いはんだしたら大丈夫になりました。
ちゃんと動いてすごくテンションが上がりました。

写真をみてもおわかりのとおりはんだづけが下手すぎます、、
子供にも馬鹿にされ（苦笑）これで動かなかったらもうはんだ付けの趣味はやめようかなとさえ思ったので、趣味をやめるようにならなくて良かったです！
こんなレベルの私でもできたよということがどなたかの励みになればと思っています。

このXIAOGYANを活用してまた何かを作りたいなと思います。
ありがとうございました。

![PXL_20230610_065827114](https://github.com/algyan/XIAOGYAN/assets/72242005/a42c8845-f482-4705-8a07-53d87591360d)

**Reply by matsujirushi** [reply]
すべて動いてよかったですー。すばらしい。

> このXIAOGYANを活用してまた何かを作りたいなと思います。

楽しみにしています😊

**Comment by uecken** [comment]
ALGYANイベントでみんなではんだ付けするものかと思って期待してました！
が違ったようで一昨日基板が届きました。

焼きXIAOにしてみました。
抵抗達が引っ付いてしまいましたが、調整して無事動きました。
初めて使う部品があって参考になりました。(7セグLED、ロータリーエンコーダ、可変抵抗、アンプ)

https://github.com/algyan/XIAOGYAN/assets/5786149/e0e120a5-2a68-4f6b-88bd-6e3cd708b06a


p.s.ハンダ付け/基板作成ありのハッカソンは聞いたことが無いなと思いまして、あったら楽しそうに思います。

**Reply by matsujirushi** [reply]
「ALGYANイベントでみんなではんだ付け」も考えていますが、みなさん既に作っちゃっているんですよねｗ
ご自身のペースで楽しんでいただければと思います！

**Comment by shoko93** [comment]
はんだ付け難しかったですが、無事できました。LEDマトリックスのところで少し不具合はありますが、他は問題なさそうです。
早速Groveセンサーも繋いでAzure IoT Centralへの接続も試してみました。

https://github.com/algyan/XIAOGYAN/assets/9836975/452bb477-aac3-4260-990e-e5a5895ee593

![IMG_0401](https://github.com/algyan/XIAOGYAN/assets/9836975/0d1317d4-0f09-4c74-8d72-9e58d9a6f2a7)
<img width="374" alt="スクリーンショット 2023-06-10 23 31 11" src="https://github.com/algyan/XIAOGYAN/assets/9836975/adb97488-0e45-48ba-af6b-700a5ba81d6b">
<img width="537" alt="スクリーンショット 2023-06-10 23 31 34" src="https://github.com/algyan/XIAOGYAN/assets/9836975/1341f70e-6ecc-4d74-b4f1-856de126f94b">

**Reply by matsujirushi** [reply]
お、IoTCいいですね。
どこのコードを参考にしていますか？？

**Reply by matsujirushi** [reply]
LEDマトリックス不調は、ここかなぁ。（写真でみる限り）

![image](https://github.com/algyan/XIAOGYAN/assets/8079404/800f29c7-a203-4d37-8455-23f765d692e5)

**Reply by shoko93** [reply]
@matsujirushi 
LEDマトリックス指摘の箇所いじったら直りました！ありがとうございます！！
IoTCですが、参考というかほぼ[この](https://github.com/Azure/azure-sdk-for-c-arduino/tree/main/examples/Azure_IoT_Central_ESP32)サンプルソースのままです。もう少しコード整理して気温以外の値も送れるようにしてみようと考えてます。

**Reply by matsujirushi** [reply]
コード整理の参考に、、、ならないと思いますが、わたしが試したときの雑記を貼っておきますｗ
https://matsujirushi.hatenablog.jp/archive/category/Azure%20IoT%20Central

**Comment by lipoyang** [comment]
XIAO ESP32「S3」で音声合成エンジンAquesTalk ESP32 (いわゆる「ゆっくりボイス」のESP32版) を無理やり動かしてみました。

https://github.com/algyan/XIAOGYAN/assets/2517317/29315f1e-32fc-4ddb-be13-4073ca5d1ca0

- 評価版のためナ行とマ行はすべてヌと発声される。(ライセンスは¥1,980@1デバイス)
- PCMデータを16bit→8bitに落としてXIAOGYANライブラリのSpeaker2で再生。
- AquesTalkのライブラリは無印ESP32用のバイナリでの配布のため、ESP32S3でもちゃんと動くのか不明。
    - 無印ESP32のコアは Xtensa LX6、ESP32S3のコアは Xtensa LX7。
    -  ESP32C3は RISC-V なので不可。

**Comment by lipoyang** [comment]
XIAO ESP32「S3」でインターネットラジオができました。
粗削りのサンプルコードですが、下記で公開しています。
https://github.com/lipoyang/XiaogyanRadio

https://github.com/algyan/XIAOGYAN/assets/2517317/1a63b7f0-f4f3-4d0f-bcbb-4fef29d134e5

**Comment by yamati-kz** [comment]
[LPS25HB使用/気圧センサーモジュールDIP化キット](https://akizukidenshi.com/catalog/g/gK-13460/)をユニバーサル部分に載せてみました。
ユニバーサル部分を使ったI2C通信に問題ないようです。
プルアップ抵抗は、LPS25HBモジュール側の[10kΩ](https://akizukidenshi.com/download/ds/akizuki/ae-lps25hb.pdf)を使用。
モジュールへの電源供給は、3.3V.
[サンプルプログラム](https://akizukidenshi.com/download/ds/akizuki/ae-lps25_sample.zip)の中のLPS25_I2C_SIMPLE20180714.inoを使用(Arduino IDE 2.1.1).
気温値を変換する際に元のソースに以下を追加すると良いようです。2の補数と変数のバイト長の関係かと思われます。
  // 極性判断 (温度)
  if ( idata >= 32768 )
    idata -= 65536;

温度値変換式は、[Technical note](https://www.st.com/resource/en/technical_note/tn1228-how-to-interpret-pressure-and-temperature-readings-in-the-lps25hb-pressure-sensor-stmicroelectronics.pdf)が重要とネット上で散見されます。

![F0ef3FtaYAA8MDc](https://github.com/algyan/XIAOGYAN/assets/135952435/9afd84e3-edde-4ccd-a29a-9233158e4458)
![lps25hb](https://github.com/algyan/XIAOGYAN/assets/135952435/ea1be814-9875-45b2-8e3f-2402e66a41bd)

**Reply by matsujirushi** [reply]
ユニバーサルの利用、初めてみました！
動いてよかったです。

idataがint型で、Arduino Uno(AVR)の場合とバイト長が違うからと思われます。（想像）
```
  int idata;
```

**Comment by yamati-kz** [comment]
[ロータリエンコーダEC12PLLVF(P-05767)/LED色(黄緑)](https://akizukidenshi.com/catalog/g/gP-05767/)を載せてみましたが、かなり見づらく感じました。軸の真上からは光っているところが視認できますが、横からだとみえないというか。

試された方がいらっしゃいましたらご感想をお聞きできたらなと思います。

マイコンでは試していませんが安定化電源に接続、組立マニュアル通りのR5＝10kΩ, R6＝10kΩの並列接続に対して、さらに10kΩ を並列に加えると少し改善する感じです。つまみはつけてません。
![F4yB_XabgAE9nZO](https://github.com/algyan/XIAOGYAN/assets/135952435/b47df34a-68e6-4b81-8dad-d0fa4f45b54e)

**Reply by cubicrootk** [reply]
私が買ったのは赤色で　手持ちの220Ωを並列に追加して、横からも見える光量で動作しました。

EC12PLLVF (LED付ロータリーエンコーダ)の仕様書は中国語なので推定になりますが、
直流電順電壓 Forward voltage(V)   紅色　Red color 1.8V
3.3Vから1.8vを差し引き、計算上、(3.3 - 1.8)/ 220 Ω = 6.8 mA となります。

草綠色 LawngreenのVfv = 2V とあるので、例えば　(3.3 - 2) / 100 = 13 mA くらいは
ESP32C3のGPIOが引き込む事が出来ると推定できます。
（ESP32C3 datasheet によると Iol 28 mA (typ) と記載されています）

**Reply by yamati-kz** [reply]
@cubicrootk  さん
詳しく解説していただきまして、ありがとうございます。
LEDの光り方具合と電流値の関係が良くわかってなくて、とても参考になりました。
試してみたいです。

**Comment by kurokouji** [comment]
XIAOGYAN基板実装しました、ユニバーサル部分とても便利です。
XIAO ESP32C3付属の外部アンテナの取付が悩みの種でした。
技適にも対応済みのXIAO ESP32C3用のミニアンテナを発見したのでXIAOGYANに縛り付けてみました。
ユニバーサルのスルーホールに細めの単線を通しアンテナがガタつかないように裏でねじるだけです。
温度・湿度・気圧が測定できるセンサもレゴバーツを利用して取付、見映えも良く着脱簡単！
![001](https://github.com/algyan/XIAOGYAN/assets/933671/d2be0980-8e5b-4c98-9c5d-775e9518c25c)

アンテナは自由に向きを変えられ、コンパクトに収納することもできます。
![002](https://github.com/algyan/XIAOGYAN/assets/933671/2fdfe01e-5697-4dcf-a41b-e5d1122ab75b)

**Comment by matsujirushi** [comment]
> ユニバーサルのスルーホールに細めの単線を通しアンテナがガタつかないように裏でねじるだけです。

ユニバーサル部の有効活用ですね👍

**Comment by cubicrootk** [comment]
Leafony(リーフォニー)という小型のプラットフォームを使っています。
   https://docs.leafony.com/docs/overview/

ハードウェアとしては Arduino Uno R3 に近くて、XIAOと比較すると
　デジタルI/O数：16/ 7 本（I2C含む）
　アナログI/O数： 6/ 4 本　です。

LeafonyのMPUとしては、ATM328P、STM32L432 、ESP32(WROOM-32)がラインアップされています。
Leafonyのセンサーとして、4-Sensors（温湿度、照度、加速度、I2C接続）等があります。

個人的にLeafonyとXIAOの連携基板を作りました。
XIAOGYANと組み合わせが出来ます。

![Leafony_on_XIAOGYAN](https://github.com/algyan/XIAOGYAN/assets/8497672/23c579b5-2468-4082-8d01-db6f80d86093)

この写真の右上の青い基板が、自作の「Leafony-XIAO連携基板」でこの基板の左端の29個の金メッキランドに、緑色のコネクタを取り付けて、Leaf（リーフォニー互換のボードの呼び方）をスタックして、ねじ止めする仕組みです。
Leafonyスタックを14ピンDIPピンヘッダーでXIAOGYANのMPUの替りに差し込んで使う訳です。

**Comment by y-ohe** [comment]
3Dプリンタでケースを作ってみました。
まだ荒削りで、ガタつくところがあったりしますが、このケースにはXIAO用のロッドアンテナを嵌められます。
先日のNT名古屋で展示してもらったので、見学された方もいるかと思います。

あと基板の固定にM3(5mm)のネジが6コ追加で必要になります。

@matsujirushi さん
ケースのデータは「case」のフォルダを作って登録するのが良いですかね？

![image](https://github.com/algyan/XIAOGYAN/assets/18491913/ab77cebb-375d-4576-aa6d-04a09e4158d8)

**Reply by 610t** [reply]
欲しい…

**Reply by y-ohe** [reply]
少しだけお待ちください。
もうしばらくしたら、完成度を上げたSTLとかのデータをこのGithubにアップします。
現物が欲しいというのであれば、基板と同じようにboothで送料程度で販売ですかね。

**Comment by matsujirushi** [comment]
@y-ohe 
3D CADは何を使っていますか？
`/enclosure/`配下にSTLとSTEPを置いて、Pull requestを発行願います。🙇‍♂️
（その際、`Allow edits by maintainers`にチェックしておいてください。）

**Reply by y-ohe** [reply]
Fusion 360です。
f3dもいりますかね。
あと、3DPDFも置くつもりです。

**Reply by matsujirushi** [reply]
f3d、step、stlをお願いします。

* 元 ... f3d
* Fusion360以外の3D-CAD向け ... step
* 3Dプリント向け ... stl

**Comment by velheor666** [comment]
Hello! I connected the accelerometer and it works on sparkfun's sketch! I have a project in which the arduino, via the SPI bus, alternataly reads data from four adxl345 accelerometers and builds 4 graphs. Could you tell me how to set up a sketch for this task?

**Reply by cubicrootk** [reply]
I suppose you could upload your sketch to your Github then you might share the link with us.

**Reply by matsujirushi** [reply]
Hi @velheor666 ,
Thanks for share the information!

Please write some more.
1. Which XIAO do you use?
2. How are the three ADXL345 wired to XIAOGYAN?

**Comment by cubicrootk** [comment]
XIAOGYAN Examples に 加速度センサーの項目がありましたが、別のIMU品種でスケッチを自作してみました。
![202312_LSM6DS3_on_XIAOGYAN](https://github.com/algyan/XIAOGYAN/assets/8497672/3973d4e2-a316-46fd-a0d8-a9503eebfcb0)

Seeed社のLSM6DS3 ボードが欠品中だったので、Adafruit社のLSM6DS3TR-C ボードを調達しました。
GroveコネクタからQwiicコネクタに変換するため、Seeed社のGrove-Qwiic Hubボードで中継しました。
（XIAOGYANは3.3V系なので、変換ケーブルを自作すれば直接接続できます。)
Grove-Qwiic Hubは５VのGroveから電源も信号線レベルも3.3Vに変換するので、M5Stack と接続する時に便利。
![LSM6DS3_Seeed_Adafruit](https://github.com/algyan/XIAOGYAN/assets/8497672/270b918b-98c0-4e61-9a6c-3eec06cff9d4)

[動作の様子](https://www.youtube.com/watch?v=zCF_jhtfxio)

**Comment by lipoyang** [comment]
XIAOGYANを基板CAD Eagle専用のグリッド切り替え用入力デバイスにしてみました。
USBキーボード or BLEキーボードにしてコマンドを送っています。
XIAO nRF52840を使用してます。GitHubで公開しました。
https://github.com/lipoyang/GridSwitch

https://github.com/algyan/XIAOGYAN/assets/2517317/13a5e9b4-7894-4a68-afc1-f0c5155f2c51

**Reply by matsujirushi** [reply]
いいですね！
[READMEに追記](https://github.com/algyan/XIAOGYAN/blob/main/README.md#%E4%BA%8B%E4%BE%8B)しました！

**Comment by kurokouji** [comment]
![image](https://github.com/algyan/XIAOGYAN/assets/933671/d7299a80-d86c-4f5d-9926-27f16f76bef4)
XIAGYAN基板を改造してST7789LCDを接続してみました。スプライト用のメモリに余裕が欲しかったので、Xiao ESP32S3を使用してLovyanGFXで駆動し、CPUクロックを80MHzに設定しました。その結果、最低でも10FPSでの表示が可能となりました。
しかし、WifiやBLE、LCDを同時に利用すると、Xiao ESP32S3から出力される3.3V電源周りが不安定になるようです。動作が安定しないため、対策としてユニバーサル領域にDC-DC ステップダウン パワーモジュールを増設し、3.3V電源を供給しました。結果として、Xiao ESP32S3から出力される3.3Vは使用を止めることとなりました。参考まで。

**Comment by kurokouji** [comment]
![XIAOGYAN_LVGL](https://github.com/algyan/XIAOGYAN/assets/933671/5d28666a-8af2-468f-83df-cc04a38d6b82)
XIAOGYAN基板を利用してLVGLの動作確認をしました。
Seeed Studio XIAO ESP32S3の少ないピンを利用してLCD ST7789とタッチパネル XPT2046を動作させることができました。
エンコーダやスイッチとバッティングするので未実装となりますが、
画面内に仮想のスライダ、ボタン、LEDを表示することができました。
